### PR TITLE
[Zen2] Fix CoordinatorTests

### PR DESCRIPTION
**Pull request by DaveCTurner**
Today the `CoordinatorTests` are not very reliable if two elections are scheduled concurrently. Although we expect occasional failures due to this, in fact the failures are much more common than expected due to a handful of issues. This PR fixes these issues.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java [hunk 1]
```diff
@@ -122,6 +122,10 @@ private void handleApplyCommit(ApplyCommitRequest applyCommitRequest) {
     }
 
     PublishWithJoinResponse handlePublishRequest(PublishRequest publishRequest) {
+        // overwrite local node
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java]: I'm not sure why this is needed. As we're using the proper transport layer in our tests, this should be taken care of by `PublicationTransportHandler`?

**Review by ywelsch — COMMENTED** [review]
I've left one comment, the rest LGTM

File: server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java (hunk 1 above)
Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java]: Well spotted, I hadn't realised that the transport layer should have been doing this when deserialising the request. The issue is that `DisruptibleMockTransport` doesn't de/serialize the requests before passing them on, so I've added that in defcc57.

**Review by ywelsch — APPROVED** [review]
Great!

**Pull request merged**